appservice/src/createAppService/AppInsightsCreateStep.ts

```diff
@@ -61,56 +65,91 @@ export class AppInsightsCreateStep extends AzureWizardExecuteStepWithActivityOut
             const aiName: string = nonNullProp(context, 'newAppInsightsName');
 
             try {
-                context.appInsightsComponent = await client.components.get(rgName, aiName);
-                this._usedExistingAppInsights = true;
+                context.appInsightsComponent = await client.components.createOrUpdate(
+                    rgName,
+                    aiName,
+                    {
+                        kind: 'web',
+                        applicationType: 'web',
+                        location: appInsightsLocation,
+                        workspaceResourceId: context.logAnalyticsWorkspace?.id
+                    });
             } catch (error) {
-                const pError: IParsedError = parseError(error);
-                // Only expecting a resource not found error if this is a new component
-                if (pError.errorType === 'ResourceNotFound') {
-
-                    context.appInsightsComponent = await client.components.createOrUpdate(
-                        rgName,
-                        aiName,
-                        {
-                            kind: 'web',
-                            applicationType: 'web',
-                            location: appInsightsLocation,
-                            workspaceResourceId: context.logAnalyticsWorkspace?.id
-                        });
-                } else if (pError.errorType === 'AuthorizationFailed') {
+                const pError = parseError(error);
+                if (pError.errorType === 'AuthorizationFailed') {
                     if (!context.advancedCreation) {
                         this._skippedCreate = true;
                     } else {
-                        await this.selectExistingPrompt(context);
-                        this._usedExistingAppInsights = true;
+                        ext.outputChannel.appendLog(l10n.t('Failed to create application insights "{0}".', aiName));
```

Comment:
Nit: This `Failed to create...` log says a lot of the same things as the line below it, I would just pick one of the two to display.